app/code/community/Bolt/Boltpay/Controller/Traits/ApiControllerTrait.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Bolt magento plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+/**
+ * Trait Bolt_Boltpay_Controller_Traits_ApiControllerTrait
+ *
+ * Defines generalized actions associated with API calls to and from the Bolt server
+ *
+ * @method Mage_Core_Controller_Response_Http getResponse()
+ * @method Mage_Core_Model_Layout getLayout()
+ */
+trait Bolt_Boltpay_Controller_Traits_ApiControllerTrait {
+    use Bolt_Boltpay_BoltGlobalTrait;
+    /**
+     * @var string The body of the request made to this controller
+     */
+    protected $payload;
+    /**
+     * @var string The signed payload which used the stores signing secret
+     */
+    protected $signature;
+    /**
+     * @var bool determines if JSON is expected return type for preDispatch optimization.
+     */
+    protected $willReturnJson = true;
```

Comment:
Bolt API always expect JSON I believe. do we need this?

Reply:
Yes.  There are a variety of situations, especially when thinking big picture, long-term.  Short-term, immediate, we have
- returns JSON - Web Hook, Shipping And Tax
- returns HTML - Success Page

Reply:
@daisy1754
I disagree. We must extend that class to support our API request that is outside the context of the session. That extension exhibits the exact APIController traits while it is not an API controller.  This is the exact reason traits exist.

Reply:
I'm confused. are you saying we have to extend success page controller so it can be read from Bolt's API request? I thought it's always called from user session

Reply:
It is true that the success URL is called from the user's session, however, it relies on session variables which are set during order creation.  Order creation happens outside of the user's session, so these variables are not set and available via direct call to the Magento standard class.  To support this, we must extend the method to support reading what would normally be session variables from encoded URL parameters.